src/cluster/connectionBuilder.js

```diff
@@ -1,4 +1,11 @@
 const Connection = require('../network/connection')
+const { KafkaJSConnectionError } = require('../errors')
+
+const validateBrokers = brokers => {
+  if (!brokers || brokers.length === 0) {
+    throw new KafkaJSConnectionError(`Failed to connect: expected brokers array and got nothing`)
```

Comment:
`KafkaJSConnectionError` is a retriable error (it inherits `KafkaJSError` which is retriable by default). Since retrying wouldn't change the list of brokers, this should be non-retriable.